Portable Rechargers! Neat and dangerous.

DrDiasyl opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Adds 2 types of portable rechargers into the game, backpack and belt rechargers. (Belt one has a badass variant which allows you to spin guns while wearing it, it can be crafted by mixing a belt recharger with a holster)

<details>
  <summary>Sprites</summary>

![backpack recharger](https://github.com/tgstation/tgstation/assets/42353186/48b8234f-3d11-4fa5-abdd-deda732b3e97)

![belt recharger](https://github.com/tgstation/tgstation/assets/42353186/94f3e79d-0ff6-4ea0-9796-d8bf92c3630a)

![badass recharger](https://github.com/tgstation/tgstation/assets/42353186/941e3ee4-85d6-407c-8200-ce873f1b14cd)

</details>

They work like normal rechargers, and cannot be upgraded so they're recharging always at normal speed, and they use power cells to recharge instead of the APC power.
While recharging it produces a distinct looping sound for everyone around the wearer to hear (even if the gun has fully recharged).

To function they must be worn on their respective slots, otherwise, they will refuse to accept guns.

The backpack recharger is acquired by the Weapons RnD research, the belt recharger is acquired by the Advanced Weapons RnD research.
One backpack recharger starts in every armoury with a high-capacity cell loaded in. High-capacity power cells can recharge (around) 2 guns only before going dry.

The power cells of portable rechargers can be swapped with a screwdriver.

When the wearer is hit by a projectile (with damage higher than 10) or EMP, there is a chance for it to malfunction (5% for backpack and 10% for belt) which will cause it to blare alarms violently and blow its power cell (so you can sabotage them beforehand, and also explosion size depends on the power cell) in 5 seconds so the wearer has a chance to throw it away.

<details>
  <summary>DRAMATIC SHOWCASE OF MALFUNCTION</summary>
  
https://github.com/tgstation/tgstation/assets/42353186/582c578b-6a50-4ca5-9967-9ec6cf4b5a9e
 
 </details>
 
<details>
  <summary>Actual Showcase</summary>
  
 https://github.com/tgstation/tgstation/assets/42353186/49173837-474c-49e4-ba09-daba62f622d6
 
 </details>
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
To recharge their guns where needed people right now are just dragging around a recharger and wrenching it on the floor when required, but this just looks silly.

This gives an alternative to this silly recharger dragging with those portable rechargers. The backpack recharger takes one of the most important slots of the inventory (and its production is blocked under research), while belt one is locked under rather heavy research and has a high chance of blowing up. You cannot just have one in your hand as they only work when you're wearing them, and dragging a recharger would still be more efficient.
Those portable rechargers can be funny with security-making squires who recharge their guns.

It making noise while recharging, having a chance to blow up from getting shot/EMP'd, being locked under research, still actually being recharger so you have to wait for the guns to recharge, and taking one of the most important inventory slots makes it balanced, especially when dragging a recharger is still an option which funnily works well but looks silly.

And you can actually use it blowing up against your enemy, by throwing it at them or rushing into their lines to use it as a DIY grenade 😆

Also, @Fikou thinks this is "Cool"
<!-- Argue for the merits of your changes and how they benefit the game, especially if they are controversial and/or far reaching. If you can't actually explain WHY what you are doing will improve the game, then it probably isn't good for the game in the first place. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
add: New piece of equipment, PORTABLE WEAPON RECHARGERS! 2 variants of them exist, Backpack Recharger and Belt Recharger. They use power cells to recharge guns on the go but be cautious, they tend to malfunction and explode.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

DrDiasyl committed:
Adds portable rechargers!

DrDiasyl committed:
Merge remote-tracking branch 'upstream/master' into portable-rechargers

DrDiasyl committed:
Places one backpack recharger in every armory

DrDiasyl committed:
Fixing mistakes

Striders13 commented:
imo having it explode when hit by non-emp is excessive

SimplySmiley123 commented:
The "being shot can make you explode" bit is imo overkill.
EMP is already piss-easy to get and use, no need to discourage anyone from bothering even more, especially considering nobody will use the backpack one _prolly ever_

SimplySmiley123 commented:
> The "being shot can make you explode" bit is imo overkill. EMP is already piss-easy to get and use, no need to discourage anyone from bothering even more, especially considering nobody will use the backpack one _prolly ever_ since, yeah it's gonna take the most important slot for a role that cares most about being prepared. Belt could be good, but the backpack will jsut sit there as a waste of time spent on making it.

DrDiasyl commented:
> > The "being shot can make you explode" bit is imo overkill. EMP is already piss-easy to get and use, no need to discourage anyone from bothering even more, especially considering nobody will use the backpack one _prolly ever_ since, yeah it's gonna take the most important slot for a role that cares most about being prepared. Belt could be good, but the backpack will jsut sit there as a waste of time spent on making it.

It is part of the Weapons Research, RnD gets that usually because well it has other things than just guns.
Also, all it takes to make those is just printing at techfab.

Also backpack one surely will be used sometimes, as one always spawns in armoury.

SimplySmiley123 commented:
> > > The "being shot can make you explode" bit is imo overkill. EMP is already piss-easy to get and use, no need to discourage anyone from bothering even more, especially considering nobody will use the backpack one _prolly ever_ since, yeah it's gonna take the most important slot for a role that cares most about being prepared. Belt could be good, but the backpack will jsut sit there as a waste of time spent on making it.
> 
> It is part of the Weapons Research, RnD gets that usually because well it has other things than just guns. Also, all it takes to make those is just printing at techfab.
> 
> Also backpack one surely will be used sometimes, as one always spawns in armoury.

It will be used once or twice when it gets added, and same way as the pepperspray backpack, won't be touched ever again afterwards and will be forgotten. Especially considering all the risks with randomly getting blown up because of EMP or getting shot at in any way.
Seriously, what's with the last part? Do you _really_ want anyone to bother with using these?

DrDiasyl commented:
> > > > The "being shot can make you explode" bit is imo overkill. EMP is already piss-easy to get and use, no need to discourage anyone from bothering even more, especially considering nobody will use the backpack one _prolly ever_ since, yeah it's gonna take the most important slot for a role that cares most about being prepared. Belt could be good, but the backpack will jsut sit there as a waste of time spent on making it.
> > 
> > 
> > It is part of the Weapons Research, RnD gets that usually because well it has other things than just guns. Also, all it takes to make those is just printing at techfab.
> > Also backpack one surely will be used sometimes, as one always spawns in armoury.
> 
> It will be used once or twice when it gets added, and same way as the pepperspray backpack, won't be touched ever again afterwards and will be forgotten. Especially considering all the risks with randomly getting blown up because of EMP or getting shot at in any way. Seriously, what's with the last part? Do you _really_ want anyone to bother with using these?

It doesn't blow up immediately, you have 3 seconds to throw it away which is somewhat a lot from what I tested.

DrDiasyl commented:
Also, 5% and 15% are not that big.
Did you know that if you hold the grenade there is a 15% chance it will IMMEDIATELY blow up in your hand if you get shot?

SimplySmiley123 commented:
> > > > > The "being shot can make you explode" bit is imo overkill. EMP is already piss-easy to get and use, no need to discourage anyone from bothering even more, especially considering nobody will use the backpack one _prolly ever_ since, yeah it's gonna take the most important slot for a role that cares most about being prepared. Belt could be good, but the backpack will jsut sit there as a waste of time spent on making it.
> > > 
> > > 
> > > It is part of the Weapons Research, RnD gets that usually because well it has other things than just guns. Also, all it takes to make those is just printing at techfab.
> > > Also backpack one surely will be used sometimes, as one always spawns in armoury.
> > 
> > 
> > It will be used once or twice when it gets added, and same way as the pepperspray backpack, won't be touched ever again afterwards and will be forgotten. Especially considering all the risks with randomly getting blown up because of EMP or getting shot at in any way. Seriously, what's with the last part? Do you _really_ want anyone to bother with using these?
> 
> It doesn't blow up immediately, you have 3 seconds to throw it away which is somewhat a lot from what I tested.

Still.
EMP is extremely easy to get and use and getting shot at is also, fairly common encounter.
Every single use would be a gamble that nobody in security will be willing to seriously take.

It really won't help these in "beingi nteresting" but only help these be _fucking useless._

DrDiasyl commented:
Anyway, I will wait for what maintainers have to say, if they dislike the chance to blow up - I will remove it, otherwise, it will stay.

If after this gets added it will be too excessive, anyone is free to PR lowering or remove it.

DrDiasyl committed:
Fixes inhand_icon_state names

Striders13 commented:
> Also, 5% and 15% are not that big.
> 
> Did you know that if you hold the grenade in your hand there is a 15% chance it will IMMEDIATELY blow up in your hand if you get shot?

you hold a grenade for a max of 3 seconds. This thing is always on you.

DrDiasyl commented:
> > Also, 5% and 15% are not that big.
> > Did you know that if you hold the grenade in your hand there is a 15% chance it will IMMEDIATELY blow up in your hand if you get shot?
> 
> you hold a grenade for a max of 3 seconds. This thing is always on you.

Good point, but 5% is ain't 15%, and also it doesn't blow up immediately. 15% is only given to the belt for it not taking up the most important slot.
If you're swift enough, you can use the recharger as a DIY grenade when it gets malf 😄

SimplySmiley123 commented:
> > > Also, 5% and 15% are not that big.
> > > Did you know that if you hold the grenade in your hand there is a 15% chance it will IMMEDIATELY blow up in your hand if you get shot?
> > 
> > 
> > you hold a grenade for a max of 3 seconds. This thing is always on you.
> 
> Good point, but 5% is ain't 15%, and also it doesn't blow up immediately. 15% is only given to the belt for it not taking up the most important slot. If you're swift enough, you can use the recharger as a DIY grenade when it gets malf 😄

And that 5% is on something that's practically gonna be useless.
And that 15% on the belt one is gonna make a lotta people not want to use it. Players already don't like it on grenades they hold for 2-3 seconds, what the hell would happen if it was for a piece of equipment you wear for way longer then that?

DrDiasyl commented:
> > > > Also, 5% and 15% are not that big.
> > > > Did you know that if you hold the grenade in your hand there is a 15% chance it will IMMEDIATELY blow up in your hand if you get shot?
> > > 
> > > 
> > > you hold a grenade for a max of 3 seconds. This thing is always on you.
> > 
> > 
> > Good point, but 5% is ain't 15%, and also it doesn't blow up immediately. 15% is only given to the belt for it not taking up the most important slot. If you're swift enough, you can use the recharger as a DIY grenade when it gets malf 😄
> 
> And that 5% is on something that's practically gonna be useless. And that 15% on the belt one is gonna make a lotta people not want to use it. Players already don't like it on grenades they hold for 2-3 seconds, what the hell would happen if it was for a piece of equipment you wear for way longer then that?

This is mostly because they're exploding instantly, I specifically made it so you have time to throw away the recharger if it gets malf.

Also, just don't get shot that often for that small small chance to activate.

SimplySmiley123 commented:
> > > > > Also, 5% and 15% are not that big.
> > > > > Did you know that if you hold the grenade in your hand there is a 15% chance it will IMMEDIATELY blow up in your hand if you get shot?
> > > > 
> > > > 
> > > > you hold a grenade for a max of 3 seconds. This thing is always on you.
> > > 
> > > 
> > > Good point, but 5% is ain't 15%, and also it doesn't blow up immediately. 15% is only given to the belt for it not taking up the most important slot. If you're swift enough, you can use the recharger as a DIY grenade when it gets malf 😄
> > 
> > 
> > And that 5% is on something that's practically gonna be useless. And that 15% on the belt one is gonna make a lotta people not want to use it. Players already don't like it on grenades they hold for 2-3 seconds, what the hell would happen if it was for a piece of equipment you wear for way longer then that?
> 
> This is mostly because they're exploding instantly, I specifically made it so you have time to throw away the recharger if it gets malf.
> 
> Also, just don't get shot that often for that small small chance to activate.

Try dodging projectiles in this game then. it's a gamble wether you'll actually walk past a shot or if it will edge on your tile and hit you, or if you won't just lag out and get hit anyway because you couldn't physically move fast enough.

And still, delayed or not it's still a fucking explosion on something you're gonna be sacrificing your backpack or belt for. Belt is less important, sure, but it's still important for a secoff.
It's genuinely not something that would be worth this explosion risk at all.

DrDiasyl commented:
> > > > > > Also, 5% and 15% are not that big.
> > > > > > Did you know that if you hold the grenade in your hand there is a 15% chance it will IMMEDIATELY blow up in your hand if you get shot?
> > > > > 
> > > > > 
> > > > > you hold a grenade for a max of 3 seconds. This thing is always on you.
> > > > 
> > > > 
> > > > Good point, but 5% is ain't 15%, and also it doesn't blow up immediately. 15% is only given to the belt for it not taking up the most important slot. If you're swift enough, you can use the recharger as a DIY grenade when it gets malf 😄
> > > 
> > > 
> > > And that 5% is on something that's practically gonna be useless. And that 15% on the belt one is gonna make a lotta people not want to use it. Players already don't like it on grenades they hold for 2-3 seconds, what the hell would happen if it was for a piece of equipment you wear for way longer then that?
> > 
> > This is mostly because they're exploding instantly, I specifically made it so you have time to throw away the recharger if it gets malf.
> > 
> > Also, just don't get shot that often for that small small chance to activate.
> 
> Try dodging projectiles in this game then. it's a gamble wether you'll actually walk past a shot or if it will edge on your tile and hit you, or if you won't just lag out and get hit anyway because you couldn't physically move fast enough.
> 
> And still, delayed or not it's still a fucking explosion on something you're gonna be sacrificing your backpack or belt for. Belt is less important, sure, but it's still important for a secoff.
> It's genuinely not something that would be worth this explosion risk at all. Especially considering how fucking easy it would be to cause the damn thing to explode.

Sounds like a skill issue.

Singul0 commented:
power creep is fine if its for the station

YuliceTheDruid commented:
Yeah these things suck huge eggs and I'd probably still drag a recharger around if I'm being real. Dead ass content.

DrDiasyl commented:
If you're too scared of a 5% chance, you can use them when fighting something that doesn't have guns or EMPs.

SimplySmiley123 commented:
> > > > > > > Also, 5% and 15% are not that big.
> > > > > > > Did you know that if you hold the grenade in your hand there is a 15% chance it will IMMEDIATELY blow up in your hand if you get shot?
> > > > > > 
> > > > > > 
> > > > > > you hold a grenade for a max of 3 seconds. This thing is always on you.
> > > > > 
> > > > > 
> > > > > Good point, but 5% is ain't 15%, and also it doesn't blow up immediately. 15% is only given to the belt for it not taking up the most important slot. If you're swift enough, you can use the recharger as a DIY grenade when it gets malf 😄
> > > > 
> > > > 
> > > > And that 5% is on something that's practically gonna be useless. And that 15% on the belt one is gonna make a lotta people not want to use it. Players already don't like it on grenades they hold for 2-3 seconds, what the hell would happen if it was for a piece of equipment you wear for way longer then that?
> > > 
> > > 
> > > This is mostly because they're exploding instantly, I specifically made it so you have time to throw away the recharger if it gets malf.
> > > Also, just don't get shot that often for that small small chance to activate.
> > 
> > 
> > Try dodging projectiles in this game then. it's a gamble wether you'll actually walk past a shot or if it will edge on your tile and hit you, or if you won't just lag out and get hit anyway because you couldn't physically move fast enough.
> > And still, delayed or not it's still a fucking explosion on something you're gonna be sacrificing your backpack or belt for. Belt is less important, sure, but it's still important for a secoff.
> > It's genuinely not something that would be worth this explosion risk at all. Especially considering how fucking easy it would be to cause the damn thing to explode.
> 
> Sounds like a skill issue.

Sounds like a good idea on paper but shit execution.
Good fucking luck getting anyone to use these with a chance for RNG to fuck someone over for daring to use a piece of equipment.

Fikou commented:
stop posting walls of quotes that makes stuff horrible to read to go in an argument in circles

GuillaumePrata commented:
Adding this gives us a good excuse to remove the silly draggable rechargers meta.

Good PR by this alone.

Now memery aside, the danger of blowing up can by ignored by simply hiding behind a window, this is a nice buff for the defensive playstyle of security while adding a danger for Officers that are running around in maint after antags or randomly fighting on hallways.

Exester509 commented:
I like them, though I think the explosive malfunction is overkill, especially since you're already sacrificing your backpack or belt slots in exchange for this convenience, arguably the two most important storage slots.
Maybe an emag interaction to use these as improvised grenades can work, which can also be used to dunk on that secoff when you blow him up with his own tool.

That aside, it's great to see the two base weapon researches giving something actually useful instead of further bloating the RnD tree.

SimplySmiley123 commented:
> They're funny and useful in some circumstances (specifically fighting in areas without APC power), I don't really get the hate for it. A small chance of exploding seems like a decent trade off for being able to recharge energy weapons on the go (and not requiring you to physically take the gun off your person and put it in a recharger). I think people forget that videogames are supposed to be fun and not a sweatfest to see who is the most robust, and this is a fun and soulful item.

The trade-off is overkill considering YOU ALREADY TRADE YOUR BACKPACK SLOT FOR IT or a BELT slot as a SECURITY OFFICER.
Nobody fucking likes getting randomly fucked by RNG especially if you already are to sacrifice something important to use it.

DrDiasyl commented:
> I like them, though I think the explosive malfunction is overkill, especially since you're already sacrificing your backpack or belt slots in exchange for this convenience, arguably the two most important storage slots.
> Maybe an emag interaction to use these as improvised grenades can work, which can also be used to dunk on that secoff when you blow him up with his own tool.
> 
> That aside, it's great to see the two base weapon researches giving something actually useful instead of further bloating the RnD tree.

I like the idea of them being a improvised weapon on the field, of that it getting hit while you're fighting, so you just chuck it at the shooter and make him blow up. Power cell explosions are underestimated.

Exester509 commented:
> > I like them, though I think the explosive malfunction is overkill, especially since you're already sacrificing your backpack or belt slots in exchange for this convenience, arguably the two most important storage slots.
> > Maybe an emag interaction to use these as improvised grenades can work, which can also be used to dunk on that secoff when you blow him up with his own tool.
> > That aside, it's great to see the two base weapon researches giving something actually useful instead of further bloating the RnD tree.
> 
> I like the idea of them being a improvised weapon on the field, of that it getting hit while you're fighting, so you just chuck it at the shooter and make him blow up. Power cell explosions are underestimated.

Yeah, but that's always leaving you without what you chose to bring in order to recharge your weapons. You can't consistenly rely on using it as a weapon, nor on using it as a portable recharger, which is what I think some people on the thread wanted to point out. The emag (or even a hackable wire, actually) would give you the control of whether or not you need an explosion right now, instead of the bullets that are also hitting your body

DrDiasyl commented:
It only registers the hits if it's worn, so if you know you're about to get Swiss cheesed you can take it off temporarily.

atlasle commented:
@Fikou this IS cool

DrDiasyl commented:
> > The power cells of portable rechargers can be swapped with a screwdriver, removing a power cell takes 3 seconds.
> 
> This gives me vibes back to
> 
> * [Swappable energy gun cells #78987](https://github.com/tgstation/tgstation/pull/78987)
> 
> Don't know how this one will play out

That PR was immediately closed by Fikou, while this one is deemed "Cool" by Fikou, and he said he would not close it. No idea about other maintainers.

DrDiasyl commented:
Weird that the space_turf_check fails.
I checked the coordinates, and there is no any space turfs here.

NoeSilent2 commented:
> It will be used once or twice when it gets added, and same way as the pepperspray backpack, won't be touched ever again afterwards and will be forgotten. Especially considering all the risks with randomly getting blown up because of EMP or getting shot at in any way. Seriously, what's with the last part? Do you _really_ want anyone to bother with using these?

Nobody uses pepperspray as is, thats why a pepperspray backpack wouldn't be successful. Everybody uses batons and energy guns-- to the point where they are literally moving the charging equipment around just to use it more. You don't see anybody assembling pepper spray refillers where they go. This PR is to give security an alternative to charging equipment around the station without having to unwrench and wrench things to floors-- which may be potentially unpowered floors.

As security I have never once been EMP'd (probably far more common on LRP than MRP to be fair) except for whilst fighting EMP blobs. EMP's are already powerful at disabling ALL energy weaponry, so whether your charger backpack is exploding or not your life expectancy is already down by a marginal percent. The numbers for external damage thresholds on the explosion mechanism _could_ be changed, but most of the time you'll be dead once you get enough damage to set it's not that bad numbers wise. 

Furthermore, it doesn't matter if this is not a super meta must-have pick for LRP validhunting powergamers to ALWAYS wear, it has a well fit niche of charging on the go in unpowered areas, patrolling without a charger to drag, or things like disaster relief when you have more important people and things to be moving around. It'll keep people up and mobile in fights and allow them to charge guns without leaving them out in the open for others to potentially grab.

There are people who want this, and it doesn't seem like it is going to be earth shatteringly powerful, so there is no harm in letting people try it out and rebalance it should it be too underpowered or somehow overpowered.

Additionally, for some of you, Diasyl obviously put a lot of thought and care into every inch of this, so all criticism is best served respectfully-- especially when you think the idea is good but have issue with one or two parts. Making other peoples days better is not a lot of effort, but it goes a long way, especially when they are doing all of this work for the totality of $0.

SomeoneYouProbablyKnow commented:
Why is everyone so focused on the explosion? It's a FUN.
But also you literally get a 3 second grenade when you need it most (you are getting shot at). It's rarely gonna hurt the wearer, it exists to prevent people from having endlessly recharging lasers by sometimes deleting the recharger.
But question : if you get hit and it's about to go off can you take out the battery with a screwdriver, or is it already too late?

It's a shame it's not gonna see much use since the belt version is RnD locked and until some brave soul unfucks storage no one is replacing their backpacks with this (except like 3 turbo gamers who alredy wipe the floor with most people).
Consider giving sec a belt recharger instead because once it blows up it's gone for the shift and it WILL blow up faster than any reasonable threat shows up. But maybe that's too much sec creep, god knows they have enough tools as is.

timothymtorres commented:
EMP effects come in two forms - `EMP_HEAVY` and `EMP_LIGHT`.  The heavy should cause automatic detonation.

Exester509 commented:
> Why is everyone so focused on the explosion? It's FUN. But also you literally get an emergency fuckoff grenade when you need it most (you are getting shot at). It's rarely gonna hurt the wearer unless you get stunned first or something but such is life. It exists to prevent people from having endlessly rechargable lasers by sometimes deleting the recharger.
> 
> But question : if you get hit and it's about to go off can you take out the battery with a screwdriver, or is it already too late? Do you lose the charging gun as well?
> 
> It's a shame it's not gonna see much use since the belt version is RnD locked and until some brave soul unfucks storage no one is replacing their backpacks with this (except like 3 turbo gamers who alredy wipe the floor with most people while carrying a duffel in their spare arm).
> 
> Consider giving sec a belt recharger instead because once it blows up it's gone for the shift and it WILL blow up before any other threat shows up unless the warden's got enough braincells to prevent sec tiding armoury shiftstart . But maybe that's too much sec creep, god knows they have enough tools as is, too many in fact.

I'm focused because of two main reasons

1. If I invest my time into R&D, resources into upgrading the cell and storage space by sacrificing my backpack or belt to recharge my weapons on the go, it's safe to assume I want to recharge them, not to use it as a grenade.
2. Even as a makeshift grenade, you don't know if it'll explode when you're behind cover, with all your equipment shutdown due to an EMP, or if you'll be crit from a shotgun blast with a bluespace cell ticking on your back about to gib you. That inconsistency really is what kills any possible functionality for me, there's no safe way to determine if you'll be able to use this offensively at all.

Aditionally, people are already prevented from having infinitely rechargeable lasers, this has a permanent 10% recharge rate, which is really low, and drains cells that you need to carry on yourself. It's already balanced because it's slow and with limited uses.

SomeoneYouProbablyKnow commented:
It is near infinite even if it's slow.

But yea you're correct. It's unreliable and you have to invest time to get it. You want to recharge guns but have to improvise when it inevitably starts beeping. In a ton of situations it can just fuck you over (Although arguably you are dead in the water in most of them anyway).

I just don't think that means the mechanic has to change. The reason it's not gonna see any use is not its viablity it's the fact that you have to go through RnD (spend time you would rather spend playing the game) and the fact that you have to remembe RnD exists.

Sec would glady use and abuse it if it spawned shiftstart (which is why it doesn't). And that's all there needs to be said about its power level.

DrDiasyl commented:
> It is near infinite even if it's slow.
> 
> But yea you're correct. It's unreliable and you have to invest time to get it. You want to recharge guns but have to improvise when it inevitably starts beeping. In a ton of situations it can just fuck you over (Although arguably you are dead in the water in most of them anyway).
> 
> I just don't think that means the mechanic has to change. The reason it's not gonna see any use is not its viablity it's the fact that you have to go through RnD (spend time you would rather spend playing the game) and the fact that you have to remembe RnD exists.
> 
> Sec would glady use and abuse it if it spawned shiftstart (which is why it doesn't). And that's all there needs to be said about its power level.



> Why is everyone so focused on the explosion? It's FUN. But also you literally get an emergency fuckoff grenade when you need it most (you are getting shot at). It's rarely gonna hurt the wearer unless you get stunned first or something but such is life. It exists to prevent people from having endlessly rechargable lasers by sometimes deleting the recharger.
> 
> But question : if you get hit and it's about to go off can you take out the battery with a screwdriver, or is it already too late? Do you lose the charging gun as well?
> 
> It's a shame it's not gonna see much use since the belt version is RnD locked and until some brave soul unfucks storage no one is replacing their backpacks with this (except like 3 turbo gamers who alredy wipe the floor with most people while carrying a duffel in their spare arm).
> 
> Consider giving sec a belt recharger instead because once it blows up it's gone for the shift and it WILL blow up before any other threat shows up unless the warden's got enough braincells to prevent sec tiding armoury shiftstart . But maybe that's too much sec creep, god knows they have enough tools as is, too many in fact.

When it's ticking it's over, you can't remove the cell.

Also, once its researched it's printable at the techfab, and Armory starts with a single backpack one.

DrDiasyl commented:
I don't know what the time investments really are, as if your blows up you can just print another one alongside with another power cell, its not that difficult.

SomeoneYouProbablyKnow commented:
> When it's ticking it's over, you can't remove the cell.

Good, wouldn't want people to bypass the explosion

Exester509 commented:
> I don't know what the time investments really are, as if your blows up you can just print another one alongside with another power cell, its not that difficult.

Time investment is getting weapons dev I or even II just to obtain an already situational, inconsistent recharger. If this got merged right now, I'd still choose to drag a recharger behind me because the inconsistency it provides is just not worth the extra time spent researching weapons development. If this is meant to replace that strategy, I'd suggest either rethinking or removing the explosion altogether (or make it upgradeable with capacitors with an increase in its explosion risk, that'd be pretty funny in my opinion)

SomeoneYouProbablyKnow commented:
>  If this is meant to replace that strategy, I'd suggest either rethinking or removing the explosion altogether

Nothing short of changing how current stationery rechargers work is gonna kill that strategy.

The average player will use the option that's just the least of busywork. Most of them don't move rechargers at all. 
And RnD is unfun busywork. It already has self recharging guns, not like they see much use.

DrDiasyl commented:
> > I don't know what the time investments really are, as if your blows up you can just print another one alongside with another power cell, its not that difficult.
> 
> Time investment is getting weapons dev I or even II just to obtain an already situational, inconsistent recharger. If this got merged right now, I'd still choose to drag a recharger behind me because the inconsistency it provides is just not worth the extra time spent researching weapons development. If this is meant to replace that strategy, I'd suggest either rethinking or removing the explosion altogether (or make it upgradeable with capacitors with an increase in its explosion risk, that'd be pretty funny in my opinion)

But it isn't even you who is investing time, its Science who is researching them.
You are not directly involved in them researching them, you're not investing anything.

MrMelbert commented on code/game/objects/items/defib.dm:
```diff
@@ -108,6 +108,8 @@
 		. += "[nocell_state]"
 	if(!safety && emagged_state)
 		. += emagged_state
+		.
```
?

MrMelbert commented on code/game/objects/items/devices/portable_recharger.dm:
```diff
@@ -0,0 +1,330 @@
+
+/obj/item/portable_recharger/equipped(mob/user, slot, initial)
+	. = ..()
+	if((slot & ITEM_SLOT_BACK) || (slot & ITEM_SLOT_BELT))
```
```suggestion
	if((slot & (ITEM_SLOT_BACK|ITEM_SLOT_BELT))
```
Bu also, we can just check for all valid slots 
```suggestion
	if((slot & slot_flags)
```

MrMelbert commented on code/game/objects/items/devices/portable_recharger.dm:
```diff
@@ -0,0 +1,330 @@
+	var/obj/item/stock_parts/cell/high/cell
+	///What item is being charged currently?
+	var/obj/item/charging = null
+	var/using_power = FALSE //Did we put power into "charging" last process()?
```
Make this an auto doc comment

MrMelbert commented on code/game/objects/items/devices/portable_recharger.dm:
```diff
@@ -0,0 +1,330 @@
+
+	if(istype(attacking_item, /obj/item/stock_parts/cell))
+		if(cell)
+			to_chat(user, span_warning("[src] already has a cell!"))
```
Balloon alert

MrMelbert commented on code/game/objects/items/devices/portable_recharger.dm:
```diff
@@ -0,0 +1,330 @@
+			if(!user.transferItemToLoc(attacking_item, src))
+				return TRUE
+			cell = attacking_item
+			to_chat(user, span_notice("You install a cell in [src]."))
```
Balloon alert

MrMelbert commented on code/game/objects/items/devices/portable_recharger.dm:
```diff
@@ -0,0 +1,330 @@
+		return TRUE
+
+	if(!cell.charge)
+		to_chat(user, span_notice("[src] blinks red as you try to insert [attacking_item]."))
```
Balloon alert

MrMelbert commented on code/game/objects/items/devices/portable_recharger.dm:
```diff
@@ -0,0 +1,330 @@
+	if(istype(attacking_item, /obj/item/gun/energy))
+		var/obj/item/gun/energy/energy_gun = attacking_item
+		if(!energy_gun.can_charge)
+			to_chat(user, span_notice("Your gun has no external power connector."))
```
Balloon alert

MrMelbert commented on code/game/objects/items/devices/portable_recharger.dm:
```diff
@@ -0,0 +1,330 @@
+
+/obj/item/portable_recharger/attack_hand(mob/user, list/modifiers)
+	if(loc == user)
+		if(slot_flags & ITEM_SLOT_BACK)
+			if(user.get_item_by_slot(ITEM_SLOT_BACK) == src)
+				take_charging_out(user)
+			else
+				to_chat(user, span_warning("Put the [src] on your back first!"))
+
+		else if(slot_flags & ITEM_SLOT_BELT)
+			if(user.get_item_by_slot(ITEM_SLOT_BELT) == src)
+				take_charging_out(user)
+			else
+				to_chat(user, span_warning("Strap the [src] on first!"))
+		return
```
I recognize this code - and if there's anything you shouldn't copy+paste, it's defib code
```suggestion
		if(user.get_slot_by_item(src) & slot_flags)
			take_charging_out(user)
		else
			balloon_alert(user, "equip it first!")
		return TRUE
```

MrMelbert commented on code/game/objects/items/devices/portable_recharger.dm:
```diff
@@ -0,0 +1,330 @@
+/obj/item/portable_recharger/MouseDrop(obj/over_object)
+	. = ..()
+	if(ismob(loc))
+		var/mob/M = loc
+		if(!M.incapacitated() && istype(over_object, /atom/movable/screen/inventory/hand))
+			var/atom/movable/screen/inventory/hand/H = over_object
+			M.putItemFromInventoryInHandIfPossible(src, H.held_index)
```
Fix these var names

MrMelbert replied:
Actually, instead of having this code, use `/datum/element/drag_pickup`.

MrMelbert commented on code/game/objects/items/devices/portable_recharger.dm:
```diff
@@ -0,0 +1,330 @@
+
+	cell.explode()
+
+	if(charging)
+		charging.forceMove(drop_location())
```
```suggestion
	charging?.forceMove(drop_location())
```
But also: You should implement `deconstruct()` on this item, where it dumps anything inside it on the ground, and then instead of calling `qdel(src)` call `deconstruct()`

MrMelbert commented on code/game/objects/items/devices/portable_recharger.dm:
```diff
@@ -0,0 +1,330 @@
+	else
+		user.log_message("[src] power cell detonated by a EMP", LOG_GAME)
+
+	user.visible_message(span_danger("[src] on [user] starts to fuss and blare alarms violently!"))
```
Suggestion: `audible_message` instead, as it relates to sound
```suggestion
	user.audible_message(span_danger("[src] on [user] starts to fuss and blare alarms violently!"))
```

MrMelbert commented on code/game/objects/items/devices/portable_recharger.dm:
```diff
@@ -0,0 +1,330 @@
+			malfunction(holder)
+			return
+
+	if(cell)
+		cell.emp_act(severity)
```
```suggestion
	cell?.emp_act(severity)
```

MrMelbert commented on code/game/objects/items/devices/portable_recharger.dm:
```diff
@@ -0,0 +1,330 @@
+	if (. & EMP_PROTECT_CONTENTS)
+		return
+
+	if(ishuman(loc))
+		var/mob/living/carbon/human/holder = loc
+		if(prob(malf_chance))
+			malfunction(holder)
+			return
```
IMO should be blocked with `EMP_PROTECT_SELF`

MrMelbert commented on code/game/objects/items/devices/portable_recharger.dm:
```diff
@@ -0,0 +1,330 @@
+
+	if(istype(charging, /obj/item/gun/energy))
+		var/obj/item/gun/energy/energy_gun = charging
+		if(energy_gun.cell)
+			energy_gun.cell.emp_act(severity)
```
```suggestion
		energy_gun.cell?.emp_act(severity)
```

MrMelbert commented on code/game/objects/items/devices/portable_recharger.dm:
```diff
@@ -0,0 +1,330 @@
+
+	else if(istype(charging, /obj/item/melee/baton/security))
+		var/obj/item/melee/baton/security/batong = charging
+		if(batong.cell)
+			batong.cell.charge = 0
```
```suggestion
		batong.cell?.charge = 0
```

MrMelbert commented on code/modules/research/techweb/all_nodes.dm:
```diff
@@ -1604,6 +1604,7 @@
 		"pin_testing",
 		"tele_shield",
 		"lasershell",
+		"backpack_recharger",
```
ABC order

MrMelbert commented on code/modules/research/techweb/all_nodes.dm:
```diff
@@ -1615,6 +1616,7 @@
 	prereq_ids = list("adv_engi", "weaponry")
 	design_ids = list(
 		"pin_loyalty",
+		"belt_recharger",
```
ABC order

MrMelbert commented on code/game/objects/items/devices/portable_recharger.dm:
```diff
@@ -0,0 +1,330 @@
+		update_appearance()
+	return ..()
+
+/obj/item/portable_recharger/Exited(atom/movable/gone, direction)
```
Should also implement `gone == cell`, `cell = null` in this

MrMelbert commented on code/game/objects/items/devices/portable_recharger.dm:
```diff
@@ -0,0 +1,330 @@
+	throwforce = 6
+	w_class = WEIGHT_CLASS_BULKY
+	///Our power cell
+	var/obj/item/stock_parts/cell/high/cell
```
```suggestion
	var/obj/item/stock_parts/cell/cell
```
No point in saying it MUST be a high level cell, because...

Autisem replied:
if you want it to coantain a high cell, simply initialize it like

`var/obj/item/stock_parts/cell/cell = var/obj/item/stock_parts/cell/high/cell`

MrMelbert commented on code/game/objects/items/devices/portable_recharger.dm:
```diff
@@ -0,0 +1,330 @@
+
+/obj/item/portable_recharger/loaded/Initialize(mapload)
+	. = ..()
+	cell = new(src)
```
... Just pass in the path
```suggestion
	cell = new /obj/item/stock_parts/cell/high(src)
```

MrMelbert commented on code/game/objects/items/devices/portable_recharger.dm:
```diff
@@ -0,0 +1,330 @@
+			cell = attacking_item
+			to_chat(user, span_notice("You install a cell in [src]."))
+
+	if(!cell)
```
```suggestion
	if(isnull(cell))
```

SimplySmiley123 commented:
> I don't know what the time investments really are, as if your blows up you can just print another one alongside with another power cell, its not that difficult.

Sorry but.
_It's. About. The fucking. RNG. In. This._
Everything else about it is fair, but seriously, RNG ain't the way to go about something you trade-off an important inventory slot for and gotta wait a long while to even be able to print it in the first place, and then depending on it's minerals cost it could be an even bigger pain to get.
At least you would expect the _backpack_ one to not be a gamble.

Exester509 commented:
> > If this is meant to replace that strategy, I'd suggest either rethinking or removing the explosion altogether
> 
> Nothing short of changing how current stationery rechargers work is gonna kill that strategy.
> 
> The average player will use the option that's just the least of busywork. Most of them don't move rechargers at all. And RnD is unfun busywork. It already has self recharging guns, not like they see much use.

Exactly, weapons R&D is unfun busywork because it has almost no reward, self recharging guns aren't even an upgrade considering you can't fit them in recharges and unlike other self charging options, have no concealed carry whatsoever.
Do you think weapons dev serving no purpose will change if the reward is an inconsistent, RNG based slow recharger that takes up your two most important inventory slots?

Bluedino1025 commented:
i don’t think the explosives are much of an issue, it only applies to projectiles and you only really see those with loud traitors, pirates, or nukies. In which case if they shoot you, you pry aren’t gonna survive. Plus it’s funny as hell

SomeoneYouProbablyKnow commented:
> It's. About. The fucking. RNG. In. This.
Everything else about it is fair, but seriously, RNG ain't the way to go about something you trade-off an important inventory slot

There's bad RnG and there's good RnG. This is a prime example of good RnG, you choose to use this thing and when it goes off you still have options.

> At least you would expect the backpack one to not be a gamble.

Yeah fair. That could be a nice way of distinguishing them. Backpack slot is so unreasonably valuable.

> Do you think weapons dev serving no purpose will change if the reward is an inconsistent, RNG based slow recharger that takes up your two most important inventory slots?

Don't twist my words. It won't change from this item nor will it change if this particular item was OP. No single addition is gonna make a difference. You need to change existing core system for that.

What this is is an optional funny item you can use once in a while and be amused by.

> In which case if they shoot you, you pry aren’t gonna survive.

Or in that case you kinda want a funny grenade.

Exester509 commented:
> There's bad RnG and there's good RnG. This is a prime example of good RnG, you choose to use this thing and when it goes off you still have options.
> 
I mean, I'd rather have it not fail at all so my options are a charged laser gun

SimplySmiley123 commented:
> There's bad RnG and there's good RnG. This is a prime example of good RnG, you choose to use this thing and when it goes off you still have options.
> 
> > At least you would expect the backpack one to not be a gamble.
> 
> Yeah fair. That could be a nice way of distinguishing them. Backpack slot is so unreasonably valuable.
> 

It's not the good kind of RNG here.
Again, it is items taking valuable inventory space, you would at least expect these to not randomly delete themselves via an explosion, and again, especially the damn backpack one.

I hate the idea of leaving this to a dice roll. Direct sabotage sure but a random chance for it to explode doesn't sound all that fun at all. Especially when the causes are piss-easy to be at the recieving end of. Especially the EMP.

DrDiasyl commented:
Yeah this is going in circles at this point.

I get that you hate the RNG, but I think this is funny and it takes nothing to just throw it away beside time limit as have said already.

If you're sad you lost it, you can just print another one alongside the cell.

If RNG will be rather annoying somehow,  anyone is free to PR changing the value.

_This is my final say about this, unless I update anything._

CRITAWAKETS commented:
it'd probably be better for the time limit to be 5 seconds instead of 3 since you kinda wanna give 1-2 seconds of reaction-time/lag margin, renember this is gonna happen during firefights where there's probably going to be less time to react on yourself aswell as lag.

also the belt version sucks because you can't spin guns with it, clearly making it inferior to the shoulder holsters in every way.

DrDiasyl commented:
> it'd probably be better for the time limit to be 5 seconds instead of 3 since you kinda wanna give 1-2 seconds of reaction-time/lag margin, renember this is gonna happen during firefights where there's probably going to be less time to react on yourself aswell as lag.
> 
> also the belt version sucks because you can't spin guns with it, clearly making it inferior to the shoulder holsters in every way.

You know what? I will add a cowboy version of it which allows to spin the gun 🤠

SomeoneYouProbablyKnow commented:
The 10 gallon hat recharger

DrDiasyl commented:
I actually lied, its only 10% chance for the belt recharger, not 15%.

DrDiasyl committed:
Melbert changes, Badass recharger, and longer malf time

Changed what Melbert asked me to change.

Added badass recharger which allows you to spin guns in your hand while wearing it.

Now it takes 5 seconds to explode instead of 3.

DrDiasyl committed:
Forgor icons

DrDiasyl commented:
@CRITAWAKETS I added a badass belt recharger, it makes you spin your gun like if you wear a holster

YakumoChen commented:
You can only spin ballistic guns, though....?

DrDiasyl commented:
> You can only spin ballistic guns, though....?

You can spin any gun.

YakumoChen commented:
Would be cool if instead of RNG you added stacking malfunctions that ultimately culminated in an explosion

Like the safeties disabling (making it continue to waste charge if something fully charged is in it)

Halving the charge rate

Making the battery splatter out acid and delete

Etc

The belt takes fewer EMPs to reach the detonate stage because its more volatile

DrDiasyl commented:
> Would be cool if instead of RNG you added stacking malfunctions that ultimately culminated in an explosion
> 
> Like the safeties disabling (making it continue to waste charge if something fully charged is in it)
> 
> Halving the charge rate
> 
> Making the battery splatter out acid and delete
> 
> Etc
> 
> The belt takes fewer EMPs to reach the detonate stage because its more volatile

Hmm, will think about this.

Dorsisdwarf commented:
I like the item as-is tbh. The risks of wearing it make it seem more themed towards anti-blob gear where recharging your guns is a # 1 concern

SomeoneYouProbablyKnow commented:
In all fairness this is pretty mediocre as anti blob gear, unless you're doing like a one man flank. It's slow, you have to wear it, it's only a single recharger and you will get hit by stray friendly fire. Best to just move rechargers and emitters anyway.

People are just too obsessed with belts, it's not THAT big a cost. Just put your 5 security items in your backpack, there's so much space in them. The backpack variant is a headache tho. 

It's a great item when you're on the run be it from sec or from something like revs imo. Sure it sometimes turns into a bomb but that's a simply a benefit.

Loafin34 commented:
this lets you carry 5 laser guns at once with the addition of recharger backpack. cool! please make the recharger packs passively charge the gun in your active hand, that would justify the usage of explosions and a backpack slot considering it allows you to continually lay down fire.
also make it not roundstart for sec, give them something to ask research for instead of oztek and telescopic shields

DrDiasyl commented:
> change

There is only **ONE** recharger backpack at the armory round start.

To print anymore you need Weapons Development research at least.

Exester509 commented:
> It's a great item when you're on the run be it from sec or from something like revs imo. Sure it sometimes turns into a bomb but that's a simply a benefit.

Please don't call the explosion a benefit. Something with a 10% chance to turn into an IED isn't consistent enough to provide a real advantage, and I bet my neck murphy's law will make you explode right when you ran out of ammo in your gun, or, in another words, when you could use a recharger instead of a grenade.
How people think of a 10% roll while under fire as a tactical advantage is beyond me

SomeoneYouProbablyKnow commented:
> Please don't call the explosion a benefit.

If you can't or don't want to improvise you do not have to use the item?? You clearly want the option to recharge guns on the go but don't like the downside that makes the item fun, interestihg and reasonable (nor do you want to spend time in RnD but that's understandable, purple department sucks)

If you want infinite lasers at no cost play captain and take the gun 5 minutes in like every other wet noodle of a captain.

MrMelbert commented on code/game/objects/items/devices/portable_recharger.dm:
```diff
@@ -0,0 +1,337 @@
+
+/obj/item/portable_recharger/equipped(mob/user, slot, initial)
+	. = ..()
+	if(slot & (ITEM_SLOT_BACK|ITEM_SLOT_BELT))
```
```suggestion
	if(slot & slot_flags)
```
This will automatically pass in back if it's the back one, or belt if it's the belt one

MrMelbert commented on code/game/objects/items/devices/portable_recharger.dm:
```diff
@@ -0,0 +1,337 @@
+		RegisterSignal(user, COMSIG_HUMAN_CHECK_SHIELDS, PROC_REF(on_attacked))
+	else
+		UnregisterSignal(user, COMSIG_HUMAN_CHECK_SHIELDS)
+		return
```
```suggestion
```

DrDiasyl replied:
This is required for the badass recharger gun spin

MrMelbert commented on code/game/objects/items/devices/portable_recharger.dm:
```diff
@@ -0,0 +1,337 @@
+	cell.explode()
+
+	charging?.forceMove(drop_location())
+	qdel(src)
```
I still recommend implementing `deconstruct()` over doing this

```dm
/obj/item/portable_recharger/deconstruct(dissassembled)
	charging?.forceMove(drop_location())
	return ..()
```
```dm
	cell.explode()
	deconstruct()
```



As it stands if your item is destroyed through other ic means, like poured acid over or something, the gun will be deleted instead of dropping

MrMelbert commented on code/game/objects/items/devices/portable_recharger.dm:
```diff
@@ -0,0 +1,337 @@
+
+/obj/item/portable_recharger/belt/badass/equipped(mob/user, slot, initial)
+	. = ..()
+	ADD_TRAIT(user, TRAIT_GUNFLIP, CLOTHING_TRAIT)
```
This gives the trait if it's held in hands, not sure if you care

DrDiasyl replied:
Because there is a return at the <equipped> for the base portable recharger, it is only when worn

zxaber commented:
My only concern is the constant looping sound getting annoying for everyone around. It'd be one thing for it to sound only while charging, but playing the sound always seems excessive. I haven't heard it yet, though, so maybe it's not so bad.

carlarctg commented:
This is such a genius idea. The maints are heavily against portable recharging of eguns for very good desgn reasons, but this PR navigates them all perfectly. It goes one step further in making security more like Combine goons, it still limits them with the slow recharge and power cell limits. It's less of a meta solution and more of a fun gimmick, as it should be. 

How do the bag and holster interact with bulky and huge guns like the ion rifle?

I'd recommend making the explosions leave behind a damaged belt that can be repaired, so it's not just gone. Also it setting you on fire is pretty miserable, fire sucks.

DrDiasyl commented:
> This is such a genius idea. The maints are heavily against portable recharging of eguns for very good desgn reasons, but this PR navigates them all perfectly. It goes one step further in making security more like Combine goons, it still limits them with the slow recharge and power cell limits. It's less of a meta solution and more of a fun gimmick, as it should be.
> 
> How do the bag and holster interact with bulky and huge guns like the ion rifle?
> 
> I'd recommend making the explosions leave behind a damaged belt that can be repaired, so it's not just gone. Also it setting you on fire is pretty miserable, fire sucks.

It can hold any rechargeable gun into it, even huge and bulky ones.

The user gets set on fire because it blows up the cell in the recharger, and the cell has a rather big flame zone. The only way to prevent it from getting on fire is coding in automatically extinguishing the user but I think that's too gimmicky

Loafin34 commented:
can you use this with the hybrid turret gun? (one that ai turrets drop) because it would make sense

DrDiasyl commented:
> can you use this with the hybrid turret gun? (one that ai turrets drop) because it would make sense

You can't fire it... why you would ever recharge it??

SimplySmiley123 commented:
> > Please don't call the explosion a benefit.
> 
> If you can't or don't want to improvise you do not have to use the item?? You clearly want the option to recharge guns on the go but don't like the downside that makes the item fun, interestihg and reasonable (nor do you want to spend time in RnD but that's understandable, purple department sucks)
> 
> If you want infinite lasers at no cost play captain and take the gun 5 minutes in like every other wet noodle of a captain.

The downside doesn't sound like it makes the item fun but a gamble instead.

If it was anything but the fucking RNG I wouldn't be so mad about it, hell would of liked it.
But fuck the RNG part, it sucks.

SomeoneYouProbablyKnow commented:
> But fuck the RNG part, it sucks.

We've been going in circles for literal days now. I don't think i can convince you. 

I really like the design it's a shame you don't see it the same way. 
Thankfully the item is in no way mandatory (or strong enough to be ze meta) so you don't have to touch it if you don't want to (untill i chuck a beeping belt at you, that is)

DrDiasyl committed:
Update portable_recharger.dm

SimplySmiley123 commented:
> > But fuck the RNG part, it sucks.
> 
> We've been going in circles for literal days now. I don't think i can convince you.
> 
> I really like the design it's a shame you don't see it the same way. Thankfully the item is in no way mandatory (or strong enough to be ze meta) so you don't have to touch it if you don't want to (untill i chuck a beeping belt at you, that is)

I see the potential of it and the RNG element ruins it entirely to me.
Nobody likes getting fucked by RNG.

Wallemations reviewed (approved):
Having both a backpack & belt variant seems like a bit much but my job is just to look over the sprites and they look fine so

Overuseddart commented:
> I see the potential of it and the RNG element ruins it entirely to me. Nobody likes getting fucked by RNG.

Crew is already strong against antags and this makes them incredibly strong even with the downside, don’t think no downside is a good idea for such a strong item.

deathrobotpunch commented:
could you at least do a private server test on people wearing the recharger backpack to see how the RNG compares to an actual situation?

DrDiasyl commented:
> could you at least do a private server test on people wearing the recharger backpack to see how the RNG compares to an actual situation?
> 
> 

RNG isn't something you can fairly compare, because it is well random. 

But on what I did, it takes a lot of EMPs.

deathrobotpunch commented:
oh it takes EMP? well that’s fine then I assumed it was like 5 precent chance of exploding per bullet hit

DrDiasyl commented:
> oh it takes EMP? well that’s fine then I assumed it was like 5 precent chance of exploding per bullet hit

Projectile (with damage above 10) hit and EMP.

DrDiasyl committed:
Merge remote-tracking branch 'upstream/master' into portable-rechargers

projectkepler-ru commented:
This whole thread reminds me of the colt dragoon during the civil war

"So we made these powerful hand cannon but they occasionally explodes"


and the army just said We ball, ordered 1000 of those things

DrDiasyl committed:
Merge remote-tracking branch 'upstream/master' into portable-rechargers

Ghommie commented on code/game/objects/items/defib.dm:
```diff
@@ -109,6 +109,7 @@
 	if(!safety && emagged_state)
 		. += emagged_state
 
+
```
```suggestion
```

Ghommie commented on code/game/objects/items/devices/portable_recharger.dm:
```diff
@@ -0,0 +1,336 @@
+
+	if(ishuman(loc))
+		var/mob/living/carbon/human/holder = loc
+		if(prob(malf_chance) || !EMP_PROTECT_SELF)
```
```suggestion
		if(prob(malf_chance) || !(. && EMP_PROTECT_SELF))
```

Ghommie reviewed (changes requested):
I'll go ahead and merge it soon, just got a couple nits and there're conflicts.

DrDiasyl commented:
> I'll go ahead and merge it soon, just got a couple nits and there're conflicts.

Darn, I cannot access my PC for the remaining of this week so I can't fix them right now.

DrDiasyl committed:
Update code/game/objects/items/devices/portable_recharger.dm

Co-authored-by: Ghom <42542238+Ghommie@users.noreply.github.com>

DrDiasyl committed:
Update code/game/objects/items/defib.dm

Co-authored-by: Ghom <42542238+Ghommie@users.noreply.github.com>

DrDiasyl committed:
Merge remote-tracking branch 'upstream/master' into portable-rechargers

DrDiasyl committed:
fixes merge conflict

DrDiasyl commented:
@Ghommie SIKE.
I actually fixed it, it's good to go (I think)

DrDiasyl commented:
Okay, somehow I made it worse.
Maybe editing maps thru code isn't best idea

DrDiasyl committed:
FIXED, I think

DrDiasyl committed:
Revert "FIXED, I think"

This reverts commit adfe8d0f644853425c5401c2db47756e204fec80.

DrDiasyl committed:
Revert "fixes merge conflict"

This reverts commit a1a98c97a89ab0c69fd6e4daf65effb841e42926.

DrDiasyl committed:
Revert "Merge remote-tracking branch 'upstream/master' into portable-rechargers"

This reverts commit b9f67ae452231ec32d58e267f6ca303ab1e183aa, reversing
changes made to d855a1daeb1225c17bdbe6296034602c599b2465.

DrDiasyl committed:
Revert "Revert "Merge remote-tracking branch 'upstream/master' into portable-rechargers""

This reverts commit aebd2618451a35329cd4c69d3fabc58753ffa93d.

DrDiasyl committed:
Merge remote-tracking branch 'upstream/master' into portable-rechargers

DrDiasyl committed:
Incident

DrDiasyl commented:
@Ghommie NOW everything is alright.

I have no idea how I overcomplicated the solving of that little merge conflict, but now its gone and checks are passing, this is what matters.

LemonInTheDark commented:
just to clarify the point of energy weapons is you can't one man with them, you need to step back, recharge, and then go back in.
does this pr fuck w that?

DrDiasyl commented:
> just to clarify the point of energy weapons is you can't one man with them, you need to step back, recharge, and then go back in.
> does this pr fuck w that?

I put all of the points in PR body already.

- It recharges slowly so you're unarmed for long time.
- It takes one of your most important slots of BACKPACK and BELT
- Getting shot or EMP'd while wearing it has a chance to make it explode
- Only one backpack recharger spawns at armory (which can also get destroyed), to get anymore you need to get a Weapons Research which costs a lot of science points
- It produces noise when recharging a gun so your enemy can hear the moment you're vulnerable

So while you're recharging your gun or just wearing this thing you're rather vulnerable.

It has enough of disadvantages to be balanced and not shatter entirety of energy weapons design.

Ghommie commented on .vscode/settings.json:
```diff
@@ -43,5 +43,6 @@
 	},
-	"Lua.diagnostics.enable": false
+	"Lua.diagnostics.enable": false,
+	"editor.inlineSuggest.showToolbar": "onHover"
```
Is this change to the vs settings intentional? If so, shouldn't be in its own PR?

DrDiasyl replied:
Whoops, this unintentional

DrDiasyl committed:
removes unintendend change

DrDiasyl committed:
Merge remote-tracking branch 'upstream/master' into portable-rechargers

DrDiasyl committed:
CONFLICT BEGONE

DrDiasyl committed:
BEGOOOOOOONE

Ghommie commented:
I suggest you discard the map changes and redo them. It's all messed up.

...Or maybe you could put the charger in the same closet of the buckshot ammo and rename the closet if you don't want to go through it for the upteenth time?

DrDiasyl committed:
Merge remote-tracking branch 'upstream/master' into portable-rechargers

DrDiasyl committed:
no map changes for me

DrDiasyl commented:
> I suggest you discard the map changes and redo them. It's all messed up.
> 
> ...Or maybe you could put the charger in the same closet of the buckshot ammo and rename the closet if you don't want to go through it for the upteenth time?

How the hell I cancel map changes?

DrDiasyl committed:
Revert "no map changes for me"

This reverts commit 824934cb54cc8697875ed3635d55c8e6ef1ad286.

DrDiasyl committed:
Revert "BEGOOOOOOONE"

This reverts commit 0e1f23a54682cd992a9d4e341ca1e53947adc225.

DrDiasyl committed:
Revert "CONFLICT BEGONE"

This reverts commit 3ff32aa207eff34e7f152e41f261345fc6ec6d3c.

DrDiasyl committed:
e

DrDiasyl committed:
e?

DrDiasyl commented:
@Ghommie not gonna lie, I fucked up badly and have no idea what to do now

Ghommie commented:
use [git reset](https://git-scm.com/docs/git-reset) (preferably not hard) to discard the commits, then [git revert](https://git-scm.com/docs/git-revert) the map changes and try again.

DrDiasyl commented:
> Use [git reset](https://git-scm.com/docs/git-reset) to the remote master branch (preferably mixed) to discard the local commits but not the line changes, then [git revert](https://git-scm.com/docs/git-revert) the map changes and try again.


I have no idea how to use Git Bash

DrDiasyl commented:
> So sad

I will reopen it
